src/Cooked/Skeleton.hs

```diff
@@ -350,49 +356,207 @@ instance Default TxOpts where
+      drvtTreasuryWithdrawal :: Rational
+    } ->
+    TxParameterChange
+  -- | Minimum size of the Constitutional Committee
+  CommitteeMinSize :: Integer -> TxParameterChange
+  -- | The Constitutional Committee Term limit in number of Slots
+  CommitteeMaxTermLength :: Integer -> TxParameterChange
+  -- | Gov action lifetime in number of Epochs
+  GovActionLifetime :: Integer -> TxParameterChange
+  -- | The amount of the Gov Action deposit
+  GovActionDeposit :: Integer -> TxParameterChange
+  -- | The amount of a DRep registration deposit
+  DRepRegistrationDeposit :: Integer -> TxParameterChange
+  -- | The number of Epochs that a DRep can perform no activity without losing
+  -- their @Active@ status.
+  DRepActivity :: Integer -> TxParameterChange
+  -- Reference scripts fee for the minimum fee calculation
+  -- will exist later on MinFeeRefScriptCostPerByte :: Integer -> TxParameterChange
+  deriving (Show, Eq)
+
+data TxGovAction where
+  -- If several parameter changes are of the same one, only the last
+  -- one will take effect
+  TxGovActionParameterChange :: [TxParameterChange] -> TxGovAction
+  TxGovActionHardForkInitiation :: Api.ProtocolVersion -> TxGovAction
+  TxGovActionTreasuryWithdrawals :: Map Api.Credential Api.Lovelace -> TxGovAction
+  TxGovActionNoConfidence :: TxGovAction
+  TxGovActionUpdateCommittee :: [Api.ColdCommitteeCredential] -> Map Api.ColdCommitteeCredential Integer -> PlutusTx.Rational -> TxGovAction
+  TxGovActionNewConstitution :: Api.Constitution -> TxGovAction
+  deriving (Show, Eq)
+
+data TxSkelProposal where
+  TxSkelProposal ::
+    { -- | Whatever credential will get back the deposit
+      txSkelProposalAddress :: Api.Address,
+      -- | The proposed action
+      txSkelProposalAction :: TxGovAction,
+      -- | An optional script to witness the proposal and validate it. Only
+      -- relevant for parameter changes and treasury withdrawals
+      txSkelProposalWitness :: Maybe (Script.Versioned Script.Script, TxSkelRedeemer),
+      -- | An optional anchor to be given as additional data. It should
+      -- correspond to the URL of a web page
+      txSkelProposalAnchor :: Maybe String
+    } ->
+    TxSkelProposal
+  deriving (Show, Eq)
+
+makeLensesFor
+  [ ("txSkelProposalAddress", "txSkelProposalAddressL"),
+    ("txSkelProposalAction", "txSkelProposalActionL"),
+    ("txSkelProposalWitness", "txSkelProposalWitnessL"),
+    ("txSkelProposalAnchor", "txSkelProposalAnchorL")
+  ]
+  ''TxSkelProposal
+
+simpleTxSkelProposal :: (ToAddress a) => a -> TxGovAction -> TxSkelProposal
+simpleTxSkelProposal a govAction = TxSkelProposal (toAddress a) govAction Nothing Nothing
+
+withWitness :: (ToScript a) => TxSkelProposal -> (a, TxSkelRedeemer) -> TxSkelProposal
+withWitness prop (s, red) = prop {txSkelProposalWitness = Just (toScript s, red)}
+
+withAnchor :: TxSkelProposal -> String -> TxSkelProposal
+withAnchor prop url = prop {txSkelProposalAnchor = Just url}
```

Comment:
I don't really see the point of this smart constructor and these two modifiers. I would be in favor of removing them to keep things simple, since `TxSkelProposal` has only 4 straightforward fields whose names are already very clear.

I think using the actual `TxSkelProposal` constructor (with or without named field) would me more readable (not a fan of infix modifiers especially if in practice the fields are almost never supposed to be `Nothing`/`Nothing`). I see that we avoid the `toAddress` and the `Just/Nothing` but it seems like a very tiny gain of time for the writer, given how so rarely tx have  proposals.

Of course these helpers are optional but they are the ones used in reference examples.

All in all, this is absolutely not a big deal. I just feel that this is a writing style you prefer but unnecessary addition to the API. (To be fair I am also advocating in part here too for personal preferences)

Reply:
Now that I think of it, I like the smart constructors, especially because they are useful and they make use of the Conversion type classes. I propose to leave them here and will make 2 examples in the cheatsheet to showcase both possibilities.

Reply:
Since a lot of datatypes in `Skeleton` already use existential types. Could we maybe do that with `TxSkelProposal` and replace its `Api.Address` with `a` where `a` has the `ToAddress` constraint? We would get the conversion for free without the need for `simpleTxSkelProposal`.

Reply:
I made an attempt to do that and it's in fact possible using the `deriving instance` tactic for `Eq` and `Show`. However, this breaks the building of lenses as the type now varies. I advocate to keep things the way they are now as the PR is already big enough, but possibly integrate these changes in the future if that's something that's important to you.